idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2296 @@
+      "        inequality constraints with only upper bounds:        0\n",
+      "\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "   0  0.0000000e+00 5.85e+01 1.00e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0\n",
+      "   1  0.0000000e+00 3.55e-15 8.41e+00  -1.0 5.85e+01    -  1.05e-01 1.00e+00h  1\n",
+      "\n",
+      "Number of Iterations....: 1\n",
+      "\n",
+      "                                   (scaled)                 (unscaled)\n",
+      "Objective...............:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Dual infeasibility......:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Constraint violation....:   3.5527136788005009e-15    3.5527136788005009e-15\n",
+      "Complementarity.........:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Overall NLP error.......:   3.5527136788005009e-15    3.5527136788005009e-15\n",
+      "\n",
+      "\n",
+      "Number of objective function evaluations             = 2\n",
+      "Number of objective gradient evaluations             = 2\n",
+      "Number of equality constraint evaluations            = 2\n",
+      "Number of inequality constraint evaluations          = 0\n",
+      "Number of equality constraint Jacobian evaluations   = 2\n",
+      "Number of inequality constraint Jacobian evaluations = 0\n",
+      "Number of Lagrangian Hessian evaluations             = 1\n",
+      "Total CPU secs in IPOPT (w/o function evaluations)   =      0.001\n",
+      "Total CPU secs in NLP function evaluations           =      0.000\n",
+      "\n",
+      "EXIT: Optimal Solution Found.\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "{'Problem': [{'Lower bound': -inf, 'Upper bound': inf, 'Number of objectives': 1, 'Number of constraints': 16, 'Number of variables': 16, 'Sense': 'unknown'}], 'Solver': [{'Status': 'ok', 'Message': 'Ipopt 3.13.2\\\\x3a Optimal Solution Found', 'Termination condition': 'optimal', 'Id': 0, 'Error rc': 0, 'Time': 0.07779264450073242}], 'Solution': [OrderedDict([('number of solutions', 0), ('number of solutions displayed', 0)])]}"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "solver.solve(m, tee=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2b27b6b8",
+   "metadata": {},
+   "source": [
+    "This is a good sign that the model solved optimally and a solution was found. \n",
+    "\n",
+    "**NOTE:** It is a good pratice to run the model through DiagnosticsToolbox regardless of the solver termination status. \n",
+    "\n",
+    "The next section we shall focus on testing the unit model. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c0d9a96f",
+   "metadata": {},
+   "source": [
+    "# 5. Testing\n",
+    "\n",
+    "There are typically 3 types of tests:\n",
```

Comment:
Most readers are not going to be familiar with the concept of testing, so you probably need some more introduction here explaining what it is, why it is important and how & when to run tests.